### PR TITLE
ART-3797: add detail to build-sync fail notification in case of non-v…

### PR DESCRIPTION
**Pull request by mbiarnes**
…iable stream

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

**Review by joepvd — COMMENTED** [review]
Nice!

File: pyartcd/pyartcd/pipelines/build_sync.py [hunk 1]
```diff
@@ -417,9 +439,17 @@ async def handle_failure(self):
...
+            message = yaml.safe_dump(filtered_issues)
+        else:
+            message = None
+
         slack_client = self.runtime.new_slack_client()
-        msg = f'Pipeline has failed to assemble release payload for {self.version} ' \
-              f'(assembly {self.assembly}) {fail_count} times.'
+        msg = f'Pipeline has failed to assemble release payload for {self.version}' \
+            f'(assembly {self.assembly}) {fail_count} times.' \
+            f'\n {message}'
```
[the remark this answers is not in the document]
Comment (mbiarnes) [reply pyartcd/pyartcd/pipelines/build_sync.py]: done

File: pyartcd/pyartcd/pipelines/build_sync.py [hunk 2]
```diff
@@ -417,9 +439,17 @@ async def handle_failure(self):
             raise
 
         # More than 2 failures: we need to notify ART and #forum-release before breaking the build
+        filtered_issues = self.filter_assembly_issues()
+
+        if filtered_issues['assembly_issues']:
+            message = yaml.safe_dump(filtered_issues)
```
[the remark this answers is not in the document]
Comment (mbiarnes) [reply pyartcd/pyartcd/pipelines/build_sync.py]: done - replaced by report

File: pyartcd/pyartcd/pipelines/build_sync.py [hunk 3]
```diff
@@ -395,6 +395,28 @@ async def _publish(self, filename):
...
+        file_path = f'{GEN_PAYLOAD_ARTIFACTS_OUT_DIR}/assembly-report.yaml'
+
+        # Open and load the YAML file
+        with open(file_path, 'r') as file:
+            data = yaml.safe_load(file)
+
+        filtered_issues = {'assembly_issues': {}}
+
+        if 'assembly_issues' in data:
+            for tool, issues in data['assembly_issues'].items():
```
[the remark this answers is not in the document]
Comment (mbiarnes) [reply pyartcd/pyartcd/pipelines/build_sync.py]: done
replaced by component

**Pull request merged**